### PR TITLE
MODDICORE-188 Remove pubsub client dependency

### PR DESCRIPTION
**Pull request by Abdulkhakimov**
## Purpose
Remove deprecated mod-pubsub client dependency.

## Approach

- Removed pubsub client dependency
- Implemented RestUtil
- Deprecated RestEventPublisher class

## Learning
https://issues.folio.org/browse/MODDICORE-188

### CODE REVIEW

File: src/main/java/org/folio/processing/events/services/publisher/RestEventPublisher.java [hunk 1]
```diff
@@ -36,19 +42,27 @@
...
-      PubSubClientUtils.sendEventMessage(event, params).whenComplete((published, throwable) -> {
-        if (throwable != null) {
-          LOGGER.error("Error publishing {} event", event.getEventType());
-          future.completeExceptionally(throwable);
-        } else {
-          LOGGER.info("{} event has been published", event.getEventType());
-          future.complete(event);
-        }
-      });
+      RestUtil.doRequest(params, "/pubsub/publish", HttpMethod.POST, event)
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/processing/events/services/publisher/RestEventPublisher.java:45]: So we still need endpoints from pubsub library?

Comment (Abdulkhakimov) [reply src/main/java/org/folio/processing/events/services/publisher/RestEventPublisher.java:45]: Following good practices, we need to annotate Deprecated in this release and remove this functionality in the next release.

File: src/main/java/org/folio/rest/util/RestUtil.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+    WrappedResponse(int code, String body,
+                    HttpResponse<Buffer> response) {
+      this.code = code;
+      this.body = body;
+      this.response = response;
+      try {
+        json = new JsonObject(body);
+      } catch (Exception e) {
+        json = null;
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/rest/util/RestUtil.java:51]: Need to add log here

File: src/main/java/org/folio/rest/util/RestUtil.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        }
+      }
+      if (method == HttpMethod.PUT || method == HttpMethod.POST) {
+        request.sendBuffer(Buffer.buffer(new ObjectMapper().writeValueAsString(payload)), handleResponse(promise));
+      } else {
+        request.send(handleResponse(promise));
+      }
+      return promise.future();
+    } catch (Exception e) {
+      promise.fail(e);
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/rest/util/RestUtil.java:106]: Exception Fail reason not captched in method RestUtil.doRequest(params, "/pubsub/publish", HttpMethod.POST, event).
As an option we can add log here

File: src/main/java/org/folio/rest/util/RestUtil.java (hunk 2 above)
Comment (Abdulkhakimov) [reply src/main/java/org/folio/rest/util/RestUtil.java:52]: Done

File: src/main/java/org/folio/rest/util/RestUtil.java (hunk 3 above)
Comment (Abdulkhakimov) [reply src/main/java/org/folio/rest/util/RestUtil.java:108]: Done

File: pom.xml [hunk 4]
```diff
@@ -20,14 +20,15 @@
...
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <pubsub.client.version>2.4.0</pubsub.client.version>
     <commons.lang.version>2.6</commons.lang.version>
     <logger.version>1.7.25</logger.version>
     <jackson.version>2.10.5.1</jackson.version>
     <junit.version>4.13.1</junit.version>
-    <vertx-unit.version>4.0.0</vertx-unit.version>
+    <vertx.version>4.0.0</vertx.version>
     <wiremock.version>2.27.2</wiremock.version>
+    <raml-module-builder.version>33.2.2</raml-module-builder.version>
```
Comment (Aliaksandr-Fedasiuk) [inline pom.xml:29]: Could you, please, update rmb version to 33.2.3?

File: pom.xml [hunk 5]
```diff
@@ -20,14 +20,15 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <pubsub.client.version>2.4.0</pubsub.client.version>
     <commons.lang.version>2.6</commons.lang.version>
     <logger.version>1.7.25</logger.version>
     <jackson.version>2.10.5.1</jackson.version>
     <junit.version>4.13.1</junit.version>
-    <vertx-unit.version>4.0.0</vertx-unit.version>
+    <vertx.version>4.0.0</vertx.version>
```
Comment (Aliaksandr-Fedasiuk) [inline pom.xml:27]: Please update vertx version to 4.2.2

**Review by Aliaksandr-Fedasiuk — APPROVED** [review]

File: pom.xml (hunk 5 above)
Comment (Abdulkhakimov) [reply pom.xml]: done

File: pom.xml (hunk 4 above)
Comment (Abdulkhakimov) [reply pom.xml]: done

**Review by SerhiiNosko — APPROVED** [review]

**Review by RuslanLavrov — APPROVED** [review]

**Pull request merged**